### PR TITLE
octosql: init at 0.9.3

### PR DESCRIPTION
**Pull request by arikgrahl**
###### Description of changes

> OctoSQL is a query tool that allows you to join, analyse and transform data from multiple databases and file formats using SQL.

-- https://github.com/cube2222/octosql

See https://github.com/NixOS/nixpkgs/issues/190465

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/octosql/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, buildGo118Module
+, fetchFromGitHub
+}:
+
+with lib;
+
+buildGo118Module rec {
```
Comment (marsam) [inline pkgs/tools/misc/octosql/default.nix:8]: nixpkgs uses buildGo118Module by default
```suggestion
, buildGoModule
, fetchFromGitHub
}:

buildGoModule rec {
```

File: pkgs/tools/misc/octosql/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  pname = "octosql";
+  version = "0.9.3";
+
+  src = fetchFromGitHub {
+    owner  = "cube2222";
+    repo   = pname;
+    rev    = "v" + version;
```
Comment (marsam) [inline pkgs/tools/misc/octosql/default.nix:15]:
```suggestion
    rev    = "v${version}";
```

File: pkgs/tools/misc/octosql/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    sha256 = "sha256-Y6kKYW79415nCJkcIKQjcBQiFZrRCJ8If65lV9wmNFA=";
+  };
+
+  vendorSha256 = "sha256-ukNjLk1tTdw0bwXaYAEDuHfzxHuAX1xyqRqC6wmW/H4=";
+
+  ldflags = [ "-s" "-w" "-X" "github.com/cube2222/octosql/cmd.VERSION=${version}" ];
```
Comment (marsam) [inline pkgs/tools/misc/octosql/default.nix:21]:
```suggestion
  ldflags = [ "-s" "-w" "-X github.com/cube2222/octosql/cmd.VERSION=${version}" ];
```

File: pkgs/tools/misc/octosql/default.nix (hunk 1 above)
Comment (arikgrahl) [reply pkgs/tools/misc/octosql/default.nix]: Thank you for pointing out!

File: pkgs/tools/misc/octosql/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+with lib;
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/octosql/default.nix:7]:
```suggestion
```

**Pull request merged**